scripts/test-run-nats-embedded

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+start-test() {
+    local ip=$(cat $TEST_DIR/databases/*/metadata/ip)
+    local port=$(cat $TEST_DIR/databases/*/metadata/port)
+    KINE_IMAGE=$IMAGE KINE_ENDPOINT="nats://$ip:$port?embedServer" provision-kine
```

Comment:
So just to be clear, the way to use this is to start kine with a `nats://` scheme datastore endpoint with the `embedServer` URI parameter set?

Can you also do something like `nats://127.0.0.1?embedServer`? If so, what port does it use?

If I set the `dontListen` option when using the embedded server, are the address and port still required, or can I just do `nats://?embedServer&dontListen`?

Can you add some examples that cover these different possibilities to a new markdown file in the [examples](https://github.com/k3s-io/kine/tree/master/examples) directory?

Reply:
Those are correct. There are default values, such as port 4222 and the `dontListen` example is also valid. I added some comments to the backend code, but will add examples.

Reply:
just adding myself as i want to try this out when the PR is merged..

Reply:
Added an examples doc with more information.

Reply:
Thanks @bruth wil def help